Example Health URL scheme with GET, single POST and batch POSTs

cookeac opened the pull request:
Resolves #154 
* Adds healthURLScheme.json (OpenAPI example for animal health events)
* Adds icarBatchResult.json - results for batch POSTs
* Replaces icarErrorResource.json with icarErrorWarningMessageResource.json which is used for errors but also errors and warnings in batch results.
* Adds icarBatchResultSeverityType.json to provide an enumeration for error, warning, and informational messages.

In usage:
* GET methods function as they do in exampleURLScheme.json
* Single POST methods take a single resource and return the updated resource if successful, else an array of error/warnings.
* Batch POST methods take an array of resources and return a matching array of batch results, with the original metadata for matching, the updated resource IDs (where added successfully) and/or an array of errors and warnings for each resource.

cookeac committed:
Changes to error and warning handling

Renamed icarErrorResource to icarBatchResultMessage, and added icarBatchResultSeverityType enum.

cookeac committed:
Made error handling consistent across single and batch

Batch POSTS now always return an array of icarBatchResult, each of which may contain an array of icarErrorWarningMessageResource.

Single POSTs if successful return the posted object (ID filled in), otherwise they return an errors array of icarErrorWarningMessageResource.

Sorry for renaming the file, but it helped to clarify its purpose.

cookeac committed:
Touch files to resolve conflicts

cookeac committed:
Merge branch 'master' of https://github.com/cookeac/ICAR into cookeac-master

ahokkonen commented on resources/icarErrorWarningMessageResource.json:
```diff
@@ -0,0 +1,35 @@
+{
+  "description": "An RFC7807 compliant problem response for JSON APIs.",
```
Could "icarResponseMessageResource"/"icarProblemResponseMessageResource"  be a better name for that resource? As we also have "Information" severity level defined for message type.

cookeac replied:
Excellent suggestion, thank you. I struggled with the naming of that resource. Renamed to icarResponseMessageResource.

ahokkonen reviewed (commented):
Added one comment

cookeac committed:
Renamed icarErrorWarningMessageResource

Renamed icarErrorWarningMessage Resource to icarResponseMessageResource based on feedback to PR #232

ahokkonen reviewed (approved):
Looks perfect now 👍🏻 Great work!

AndreasSchultzGEA reviewed (approved):
Hi, I had no closer look at the UrlScheme, but the other parts look fine.
